src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -150,18 +282,125 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
+            bool isTar,
+            string contentFormat,
             CancellationToken cancellationToken)
         {
-            DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
-                manifestId,
-                targetDirectory,
-                proxyUri: null,
-                minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+
+            // Throw for bunch of invalid combinations of istar and contentFormat ??
+            if (!isTar)
+            {
+                DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
+                    manifestId,
+                    targetDirectory,
+                    proxyUri: null,
+                    minimatchPatterns: null);
+
+                await dedupManifestClient.DownloadAsync(options, cancellationToken);
+            }
+            else
+            {
+                string manifestPath = Path.Combine(Path.GetTempPath(), $"{nameof(DedupManifestArtifactClient)}.{Path.GetRandomFileName()}.manifest");
+                using (var manifestStream = File.Create(manifestPath))
+                {
+                    await dedupManifestClient.DownloadToStreamAsync(manifestId, manifestStream, null, cancellationToken);
+                }
+                Manifest m = JsonSerializer.Deserialize<Manifest>(File.ReadAllText(manifestPath));
+                        
+                var processTcs = new TaskCompletionSource<int>();
+
+                using (var cancelSource = new CancellationTokenSource())
+                using (var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cancelSource.Token))
+                using (var process = new Process())
+                {
+                    process.StartInfo.FileName = isWindows ? @"C:\Program Files\7-Zip\7z.exe" : "tar";
+                    process.StartInfo.Arguments = isWindows ? $"x -si -aoa -o{targetDirectory} -ttar" : $"-xf - -C {targetDirectory}";
```

Comment:
why are we using 7zip.exe with Windows, is there a perf improvement with using 7zip.exe on TAR files versus the tar executable?

Reply:
Yup, there is a performance improvement on Win using 7z.